### DISCUSSION TITLE
Issue with closeModalWithEvents()

### DISCUSSION THREAD

**Discussion by michaeljhopkins** [wire-elements/modal#356] [Q&A]
I have this code in a modal

```
class EditAuthorizedUser extends ModalComponent {
    public function submit() {
        $this->closeModalWithEvents(['authorizedUsersUpdated']);
    }
}
```
and
```
class AuthorizedUsers extends Component {
    protected $listeners = ['authorizedUsersUpdated' => '$refresh'];
    public function render() {
        return view('view.file');
    }
}
```

When I close the modal I only see these events. What am I missing to make this method work?
<img width="362" alt="Screenshot 2023-06-08 at 1 27 02 PM" src="https://github.com/wire-elements/modal/assets/4623657/3d5739be-e909-4b88-98dd-1977aa1efde6">

**Comment by feliperafael** [comment]
I'm having the same issue. When I close the modal the event doesn't seem to fire. Did you get any solution?

**Comment by lrljoe** [comment]
Disable the destruction of the component. 

It is destroying it before the event queue (internal to the component) has had a chance to run.
In the config file, ensure its set to:

```
        'destroy_on_close' => false,
```
